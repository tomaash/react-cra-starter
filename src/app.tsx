import React, { Component, Suspense } from 'react'
import logo from './logo.svg'
import { Home } from '~/components/Home'
import { Profile } from '~/components/Profile'
import { ListPage } from '~/components/ListPage'
import { Settings } from './components/Settings'
import { AppStore } from '~/stores/AppStore'

import { createHistory, LocationProvider, Link, Router } from '@reach/router'

import { toJS } from 'mobx'
import { AppMenu } from './components/AppMenu'
import { Provider, observer } from 'mobx-react'

import { library } from '@fortawesome/fontawesome-svg-core'

import {
  faBookOpen,
  faTasks,
  faSearchDollar,
  faMoneyCheckAlt,
  faHandHoldingUsd,
  faUserFriends,
  faUser,
  faThList,
  faPlusSquare
} from '@fortawesome/free-solid-svg-icons'

library.add(
  faBookOpen,
  faTasks,
  faSearchDollar,
  faMoneyCheckAlt,
  faHandHoldingUsd,
  faUserFriends,
  faUser,
  faThList,
  faPlusSquare
)

export const history = createHistory(window as any)
const appStore = new AppStore()
appStore.setupHistory(history)

@observer
export class App extends Component {
  render() {
    return (
      <Provider appStore={appStore}>
        <LocationProvider history={history}>
          <AppMenu />
          <Router>
            <Home path='/' />
            <Profile path='profile/:username' />
            <ListPage path='list' />
            <Settings path='settings' />
          </Router>
        </LocationProvider>
      </Provider>
    )
  }
}
