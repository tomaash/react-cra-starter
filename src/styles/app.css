.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

/* Hack FontAwesome "lg" icon size to be compatible with Material UI */
.svg-inline--fa.fa-lg {
  width: 25px;
}

/* remove blue outline on clicked button */
button:focus {outline:0;}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

