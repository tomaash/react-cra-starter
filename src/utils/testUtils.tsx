import React from 'react'
import ReactDOMServer from 'react-dom/server'
const path = require('path')

export const squashHtmlTags = str => str.replace(/(<[^>]+>)+/g, '\n')

export const renderText = element =>
  squashHtmlTags(ReactDOMServer.renderToStaticMarkup(element))

export const startPolly = dirname => {
  const { Polly } = require('@pollyjs/core')
  const { setupPolly } = require('setup-polly-jest')
  const FSPersister = require('@pollyjs/persister-fs')
  const XHRAdapter = require('@pollyjs/adapter-xhr')

  Polly.register(XHRAdapter)
  Polly.register(FSPersister)

  setupPolly({
    adapters: ['xhr'],
    recordIfMissing: true,
    persister: 'fs',
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(dirname, '__recordings__')
      }
    }
  })
}

export const sleep = m => new Promise(r => setTimeout(r, m))
