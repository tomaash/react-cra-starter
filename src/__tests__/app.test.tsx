import React from 'react'
import { App } from '~/app'
import { renderText } from '~/utils/testUtils'

it('renders <App/>', () => {
  const text = renderText(<App />)
  expect(text).toMatchSnapshot()
})
