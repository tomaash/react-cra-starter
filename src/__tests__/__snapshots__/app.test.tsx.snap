// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> 1`] = `
"
MyApp
Edit 
src/app.tsx
 and save to reload.
Learn React
status bar
"
`;
